src/model/statement.js

```diff
@@ -14,72 +14,29 @@ class Statement {
   namespace: string
   timestamp: number
   body: StatementBody
-  signature: Buffer
-
-  constructor (stmt: {id: string, publisher: string, namespace: string, timestamp: number, body: StatementBody, signature: Buffer | string}) {
-    this.id = stmt.id
-    this.publisher = stmt.publisher
-    this.namespace = stmt.namespace
-    this.timestamp = stmt.timestamp
-    this.body = stmt.body
-    if (Buffer.isBuffer(stmt.signature)) {
-      this.signature = (stmt.signature: any)
-    } else {
-      this.signature = Buffer.from(stmt.signature, 'base64')
-    }
-  }
-
-  static create (
-    publisherId: PublisherId,
-    namespace: string,
-    statementBody: StatementBody | StatementBodyMsg,
-    counter: number = 0,
-    timestampGenerator: () => number = Date.now)
-  : Promise<Statement> {
-    const body = (statementBody instanceof StatementBody) ? statementBody : StatementBody.fromProtobuf(statementBody)
-    const timestamp = timestampGenerator()
-    const statementId = [publisherId.id58, timestamp.toString(), counter.toString()].join(':')
-    const stmt = new Statement({
-      id: statementId,
-      publisher: publisherId.id58,
-      namespace,
-      timestamp,
-      body,
-      signature: Buffer.from('')
-    })
-    return stmt.sign(publisherId)
-  }
-
-  static createSimple (
-    publisherId: PublisherId,
-    namespace: string,
-    statementBody: {object: string | Object, refs?: Array<string>, deps?: Array<string>, tags?: Array<string>},
-    counter: number = 0,
-    timestampGenerator: () => number = Date.now
-  ): Promise<Statement> {
-    let body
-    if (typeof statementBody.object === 'object') {
-      body = new ExpandedSimpleStatementBody((statementBody: any))
-    } else {
-      body = new SimpleStatementBody((statementBody: any))
-    }
-    return Statement.create(publisherId, namespace, body, counter, timestampGenerator)
-  }
 
   static fromProtobuf (msg: StatementMsg): Statement {
     const body = StatementBody.fromProtobuf(msg.body)
 
     const {id, publisher, namespace, timestamp, signature} = msg
-    return new Statement({id, publisher, namespace, timestamp, body, signature})
+    if (signature == null) {
+      return new UnsignedStatement({id, publisher, namespace, timestamp, body})
+    }
+
+    return new SignedStatement({id, publisher, namespace, timestamp, body, signature})
   }
 
-  static fromBytes (msgBuffer: Buffer): Statement {
-    return Statement.fromProtobuf(pb.stmt.Statement.decode(msgBuffer))
+  static fromBytes (bytes: Buffer): Statement {
+    return Statement.fromProtobuf(pb.stmt.Statement.decode(bytes))
   }
 
   toProtobuf (): StatementMsg {
-    const {id, publisher, namespace, timestamp, signature, body} = this
-    return {id, publisher, namespace, timestamp: timestamp, signature, body: body.toProtobuf()}
+    const {id, publisher, namespace, timestamp, body} = this
+    const msg: StatementMsg = {id, publisher, namespace, timestamp: timestamp, body: body.toProtobuf()}
```

Comment:
actually, I don't know why I did `timestamp: timestamp` instead of just `timestamp` - probably some earlier refactor.